choreographer/src/main/java/eu/arrowhead/core/choreographer/database/service/ChoreographerPlanDBService.java

```diff
@@ -273,10 +273,15 @@ public ChoreographerStep getStepById(final long id) {
 		Assert.notNull(plan, "Plan is null.");
 
 		try {
-			choreographerPlanRepository.refresh(plan);
-			final List<ChoreographerAction> actions = choreographerActionRepository.findByPlan(plan);
-			
+			final Optional<ChoreographerPlan> optional = choreographerPlanRepository.findById(plan.getId());
```

Comment:
input parameter should be planId (long) 
optional is a very generic name => should be plan